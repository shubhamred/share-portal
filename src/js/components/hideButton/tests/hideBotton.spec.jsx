import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import HideButton from '../hideButton';

configure({ adapter: new Adapter() });

describe('components/dropDown', () => {
  it('Renders default options', () => {
    const component = shallow(
      <HideButton />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('handles click', () => {
    const props = {
      onHideBtnClick: jest.fn(),
      visible: true,
      isSection: true
    };
    const component = shallow(
      // eslint-disable-next-line react/jsx-props-no-spreading
      <HideButton {...props} />
    );
    component.simulate('click');
    expect(props.onHideBtnClick).toHaveBeenCalledTimes(1);
  });
});
