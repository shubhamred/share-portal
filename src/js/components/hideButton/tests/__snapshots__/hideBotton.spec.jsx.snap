// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/dropDown Renders default options 1`] = `
<div
  className="hideBtn"
  onClick={[Function]}
  role="presentation"
>
  <WithStyles(ForwardRef(Switch))
    checked={false}
    color="primary"
    inputProps={
      Object {
        "aria-label": "primary checkbox",
      }
    }
    name="checkedA"
    onChange={[Function]}
    size="small"
  />
  <div
    className="hideSection"
  >
    Hidden
  </div>
</div>
`;
